src/Item/hashes.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  EntityContentItemReference,
+  EntityMetadata,
+  Hashing,
+} from 'dcl-catalyst-commons'
+import { omit } from 'decentraland-commons/dist/utils'
+import { CollectionAttributes } from '../Collection'
+import { isTPCollection } from '../Collection/utils'
+import { DCLCatalystItem, ItemAttributes, TPCatalystItem } from './Item.types'
+import { buildTPItemURN, getDecentralandItemURN, isTPItem } from './utils'
+
+const THUMBNAIL_PATH = 'thumbnail.png'
+const IMAGE_PATH = 'image.png'
+
+function buildItemEntityMetadata(
+  item: ItemAttributes,
+  collection: CollectionAttributes
+): DCLCatalystItem | TPCatalystItem {
+  const isTPEntity = isTPCollection(collection) && isTPItem(item)
+
+  if (!collection.contract_address && !isTPEntity) {
+    throw new Error(
+      "The item's collection must be published to build its metadata"
+    )
+  }
+
+  // We strip the thumbnail from the representations contents as they're not being used by the Catalyst and just occupy extra space
+  const representations = item.data.representations.map((representation) => ({
+    ...representation,
+    contents: representation.contents.filter(
+      (fileName) => fileName !== THUMBNAIL_PATH
+    ),
+  }))
+
+  const itemMetadata: DCLCatalystItem = {
+    id: isTPEntity
+      ? buildTPItemURN(
+          collection.third_party_id!,
+          collection.urn_suffix!,
+          item.urn_suffix!
+        )
+      : getDecentralandItemURN(item, collection.contract_address!),
+    name: item.name,
+    description: item.description,
+    collectionAddress: collection.contract_address!,
+    rarity: item.rarity ?? undefined,
+    i18n: [{ code: 'en', text: item.name }],
+    data: {
+      replaces: item.data.replaces,
+      hides: item.data.hides,
+      tags: item.data.tags,
+      category: item.data.category,
+      representations,
+    },
+    image: IMAGE_PATH,
+    thumbnail: THUMBNAIL_PATH,
+    metrics: item.metrics,
+  }
+
+  if (!item.rarity) {
+    delete itemMetadata.rarity
+  }
+
+  if (isTPEntity) {
+    // How is the metadata for a TP item built?
+    return omit(itemMetadata, ['collectionAddress'])
```

Comment:
I think it might be a bit less weird to add the `contractAddress` if `!isTPEntity`?

It's the same in this case, but if we had more code between lines 58 and 64, the type would be "wrong" in between right?
I mean, we're doing `collection.contract_address!` but we know that `contract_address` does not exist in one scenario

Reply:
Made two functions to build the metadata of the entities, one for the TP wearables and the other for the standard wearables.